code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingExtension.java

```diff
@@ -882,6 +898,19 @@ private Map<String, Object> getXDMSharedState(final String extensionName, final
     private boolean eventIsValid(final Event event) {
         return event != null && event.getEventData() != null;
     }
+
+    private void addPushTokenToSharedState(final String pushToken, final Event event) {
+        if (StringUtils.isNullOrEmpty(pushToken)) {
+            Log.debug(
+                    MessagingConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Push token is null or empty. Not adding the push token to the shared state.");
+            return;
```

Comment:
You'll still need to set the shared state if the push token is null or empty. Change this logic to simply not add the push token to the `newMessagingState`. The reason is if the push token is set, then removed, the shared state needs to be updated with a cleared push token to reflect the change in state.

Reply:
this null or empty check shouldn't be needed as push sync requests containing null or empty tokens are discarded. i'll remove the check.

Reply:
The null check is actually still needed because this API is called on registration and is passed the persisted push token, which may be null. Let's leave the check in.